### ISSUE TITLE
[BUG] Contradictory constraints highlighted in comments for snapshot name in CreateSnapshotRequest.java

### ISSUE THREAD

**Issue by JeffH-AWS** [opensearch-project/OpenSearch#6448]
**Describe the bug**
I'm a technical writer for the OpenSearch Project, and I'm currently working on writing upgrade and migration documentation for OpenSearch and Elasticsearch users. As part of this process, I'm taking a snapshot of the cluster state before changing things. While investigating the snapshot API, I came across an inconsistent comment in [CreateSnapshotRequest.java](https://github.com/opensearch-project/OpenSearch/blob/01466a6a04e280267f013aeb14dff1f20dc7a77a/server/src/main/java/org/opensearch/action/admin/cluster/snapshots/create/CreateSnapshotRequest.java#L66-L76):

On Line 74, the comment states:
```
must not start with underscore ('-')
```

The issue is that the example character, inside the single quotes, is a hyphen and not an underscore. While we continue to improve the documentation, we sometimes rely on reviewing the code itself, or at least comments in the code, to determine what the expected behavior is, what constraints we must observe, etc. So this "just a comment" should still be accurate.

**To Reproduce**
N/A - this is just a typo in a comment in a Java file

**Expected behavior**
The line should either read:
```
must not start with underscore ('_')
```
or...
```
must not start with hyphen ('-')
```

**Plugins**
Security

**Screenshots**
N/A

**Host/Environment (please complete the following information):**
N/A

**Additional context**
N/A

**Comment by andrross** [comment]
The service enforces that the snapshot name [not start with underscore](https://github.com/opensearch-project/OpenSearch/blob/08ac17fd7b9a604ecffee0a8e0d666f4dd372dba/server/src/main/java/org/opensearch/snapshots/SnapshotsService.java#L774-L776). So there's an opportunity for a single character change PR here :)

**Comment by Renatotmmartins** [comment]
Hey @andrross,
Can you assign that to me? It would be my first contribution! 
Thanks!

**Comment by AkM-2018** [comment]
@andrross I took up the issue as it was stagnant for sometime. Please review the PR

**Comment by AkM-2018** [comment]
This issue can be closed as PR is merged!

**Comment by andrross** [comment]
Fixed by #7554
